lib/presentation/forced_update/forced_update_model.dart

```diff
@@ -0,0 +1,104 @@
+import 'dart:io';
+
+import 'package:firebase_remote_config/firebase_remote_config.dart';
+import 'package:flutter/material.dart';
+import 'package:package_info/package_info.dart';
+import 'package:url_launcher/url_launcher.dart';
+
+class ForcedUpdateModel extends ChangeNotifier {
+  ForcedUpdateModel({@required this.context});
+
+  bool isForcedUpdateFlag = false;
+  BuildContext context;
+
+  static const String DEV_VERSION_CONFIG = "dev_app_version";
+  static const String CONFIG_VERSION = "force_update_app_version";
+
+  // AppStoreのurl
+  static const APP_STORE_URL =
+      'https://apps.apple.com/jp/app/id1546722883#?mt=8';
+
+  // google play storeのurl
+  static const PLAY_STORE_URL =
+      'https://play.google.com/store/apps/details?id=com.location.locationapp.location47';
+
+  Future<void> versionCheck() async {
+    // Flagの初期化
+    this.isForcedUpdateFlag = false;
+    // versionCode(buildNumber)取得
+    final PackageInfo info = await PackageInfo.fromPlatform();
+    int currentVersion = int.parse(info.buildNumber);
+    print("currentVersion = ${currentVersion}");
+    // releaseビルドかどうかで取得するconfig名を変更
+    final configName = bool.fromEnvironment('dart.vm.product')
+        ? CONFIG_VERSION
+        : DEV_VERSION_CONFIG;
+    print("configName = ${configName}");
+
+    // remote config
+    final RemoteConfig remoteConfig = await RemoteConfig.instance;
+
+    try {
+      // 常にサーバーから取得するようにするため期限を最小限にセット
+      await remoteConfig.fetch(expiration: const Duration(seconds: 0));
+      await remoteConfig.activateFetched();
+      int newVersion = remoteConfig.getInt(configName);
+      print("newVersion = ${newVersion}");
+      if (newVersion > currentVersion) {
+        this.isForcedUpdateFlag = true;
+      }
```

Comment:
Build Noでの判断だとiOSの申請などで、リジェクトされた場合も考慮するとVersion Noで判断した方が良い気がします。
Version Noで判断した場合のソースを記載しておきます。

```suggestion
    final PackageInfo info = await PackageInfo.fromPlatform();
    String currentVersion = info.version;
    print("currentVersion = ${currentVersion}");
    // releaseビルドかどうかで取得するconfig名を変更
    final configName = bool.fromEnvironment('dart.vm.product')
        ? CONFIG_VERSION
        : DEV_VERSION_CONFIG;
    print("configName = ${configName}");

    // remote config
    final RemoteConfig remoteConfig = await RemoteConfig.instance;

    try {
      // 常にサーバーから取得するようにするため期限を最小限にセット
      await remoteConfig.fetch(expiration: const Duration(seconds: 0));
      await remoteConfig.activateFetched();
      String newVersion = remoteConfig.getString(configName);
      print("newVersion = ${newVersion}");
      if (newVersion != currentVersion) {
        this.isForcedUpdateFlag = true;
      }
```

Reply:
version numberに修正します！
加えて、RemoteConfigにiOSとAndroidそれぞれの値を設定することにします！

Reply:
5bbec42

にて、Version Noで判断するよう修正しました！